src/overlays/actors/ovl_En_Az/z_en_az.c

```diff
@@ -39,108 +80,1762 @@ const ActorInit En_Az_InitVars = {
+                        ret = 0;
+                        break;
+                    case 0x1102:
+                        func_80A98EFC(this, play, 0x1103, 12, 10);
+                        ret = 0;
+                        break;
+                    case 0x1103:
+                        func_80A98EFC(this, play, 0x1104, 13, 9);
+                        ret = 0;
+                        break;
+                    case 0x1104:
+                        func_80A98EFC(this, play, 0x1105, 12, 10);
+                        ret = 0;
+                        break;
+                    case 0x1105:
+                        if (gSaveContext.save.weekEventReg[25] & 1) {
+                            this->getItemId = GI_RUPEE_PURPLE;
+                        } else {
+                            this->getItemId = GI_HEART_PIECE;
+                            gSaveContext.save.weekEventReg[25] |= 1;
+                        }
+                        SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 0, &this->animIndex);
+                        SubS_ChangeAnimationBySpeedInfo(&brother->skelAnime, D_80A99010, 0, &brother->animIndex);
+                        ret = 7;
+                        break;
+                    case 0x1106:
+                        gSaveContext.save.weekEventReg[24] &= (u8)~4;
+                        SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 0, &this->animIndex);
+                        SubS_ChangeAnimationBySpeedInfo(&brother->skelAnime, D_80A99010, 0, &brother->animIndex);
+                        func_80A94AB8(this, play, 0);
+                        func_80A979DC(this, play);
+                        ret = 0;
+                        break;
+                    case 0x10D5:
+                    default:
+                        SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 0, &this->animIndex);
+                        this->unk_374 |= 0x20;
+                        ret = 0;
+                        break;
+                }
+            }
+            break;
+        case TEXT_STATE_NONE:
+        case TEXT_STATE_1:
+        case TEXT_STATE_CLOSING:
+        case TEXT_STATE_3:
+            break;
+    }
+    return ret;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97114.s")
+void func_80A97114(EnAz* this, PlayState* play) {
+    EnAz* brother = this->brother;
+    s32 sp20 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Az/func_80A97274.s")
+    this->actor.flags &= ~ACTOR_FLAG_10000;
+    switch (this->actor.textId) {
+        case 0x10DA:
+        case 0x10DD:
+        case 0x10E9:
+            SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 4, &this->animIndex);
+            break;
+        case 0x10EE:
+        case 0x10F3:
+        case 0x10FB:
+        case 0x1102:
+            SubS_ChangeAnimationBySpeedInfo(&this->skelAnime, D_80A99010, 9, &this->animIndex);
+            sp20 = 1;
```

Comment:
```suggestion
            sp20 = true;
```
Could you do `sp20 = false` above?